lib/alchemy/pg_search/element_extension.rb

```diff
@@ -12,21 +12,21 @@
     }
 
   has_many :searchable_essence_texts,
-    -> { joins(:element).where(searchable: true, alchemy_elements: {public: true}) },
+    -> { where(searchable: true, alchemy_elements: {public: true}) },
     class_name: 'Alchemy::EssenceText',
     source_type: 'Alchemy::EssenceText',
     through: :contents,
     source: :essence
 
   has_many :searchable_essence_richtexts,
-    -> { joins(:element).where(searchable: true, alchemy_elements: {public: true}) },
+    -> { where(searchable: true, alchemy_elements: {public: true}) },
```

Comment:
@tvdeyen Based on this change, if we try to get searchable_essence_richtexts for any instance of Alchemy::Element, it gives
```
*** ActiveRecord::StatementInvalid Exception: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "alchemy_elements"
LINE 1: ... AND (alchemy_essence_texts.searchable = true AND alchemy_el...
                                                             ^
: SELECT  "alchemy_essence_texts".* FROM "alchemy_essence_texts" INNER JOIN "alchemy_contents" ON "alchemy_essence_texts"."id" = "alchemy_contents"."essence_id" WHERE "alchemy_contents"."element_id" = $1 AND (alchemy_essence_texts.searchable = true AND alchemy_elements.public = true) AND "alchemy_contents"."essence_type" = $2 ORDER BY "alchemy_contents"."position" DESC, "alchemy_contents"."id" DESC LIMIT $3
```
Though, this doesn't impact the search because of the SQL query that gets generated via pg_search_scope :full_text_search.

Since, there are similar associations for `searchable_essence_richtexts` and `searchable_essence_pictures`, 
I tried updating the association to 
```
  has_many :searchable_essence_texts,
    -> { joins(:element).where("alchemy_essence_texts.searchable = true AND alchemy_elements.public = true") } ,
    class_name: 'Alchemy::EssenceText',
    source_type: 'Alchemy::EssenceText',
    through: :contents,
    source: :essence
``` 
And now it responds back for this and other given associations, as well as for the search query.
I had to add the query in where clause in form of string because otherwise it was not taking the correct
params and giving error. 
Please let me know your thoughts on the same. Thanks!

Reply:
@dhiraj14 thanks for the research. If it works with the String in the `where` clause then let's use this. Alternatively we could add [`.references(:alchemy_elements)`](https://apidock.com/rails/v6.0.0/ActiveRecord/QueryMethods/references) to tell Rails to include that table in the query, but using the String method is just fine.